histmaker/fill_utils.py

```
@@ -495,3 +525,64 @@ def blind_DataFrame(df: pd.DataFrame, label_out: str, SR: list) -> pd.DataFrame:
         )
     ]
     return df
+
+
+def deltaPhi_x_y(xphi, yphi, xpt, ypt):
+
+    # cast inputs to numpy arrays
+    xpt = np.array(xpt)
+    ypt = np.array(ypt)
+    xphi = np.array(xphi)
+    yphi = np.array(yphi)
+
+    x_v = vector.arr({"pt": xpt, "phi": xphi})
+    y_v = vector.arr({"pt": ypt, "phi": yphi})
+
+    signed_dphi = x_v.deltaphi(y_v)
+    abs_dphi = np.abs(signed_dphi.tolist())
+
+    # deal with the cases where phi was initialized to a moot value like -999
+    abs_dphi[(xphi > 2 * np.pi)] = -999
+    abs_dphi[yphi > 2 * np.pi] = -999
+    abs_dphi[xpt < -2 * np.pi] = -999
+    abs_dphi[ypt < -2 * np.pi] = -999
+
+    return abs_dphi
+
+
+def balancing_var(xpt, ypt):
+
+    # cast inputs to numpy arrays
+    xpt = np.array(xpt)
+    ypt = np.array(ypt)
+
+    var = np.where(ypt > 0, (xpt - ypt) / ypt, np.ones(len(xpt)) * -999)
+
+    # deal with the cases where pt was initialized to a moot value (-999 only fow now)
+    var[xpt == -999] = -999
+    var[ypt == -999] = -999
```

This is fine for now, but there can be edge cases that get ugly, when setting floats to bogus values and then checking for equality, since floats can be off by epsilon and then fail the equality check. Python usually handles this better than C++ at least...

---

Would it be easier to just check for xpt > 0?